src/hip_module.cpp

```diff
@@ -386,6 +386,113 @@ hipError_t hipExtLaunchMultiKernelMultiDevice(hipLaunchParams* launchParamsList,
     return ihipLogStatus(result);
 }
 
+namespace {
+// kernel for initializing GWS
+// nwm1 is the total number of workgroups minus 1 and rid is the GWS resource id
+__global__ void init_gws(uint nwm1, uint rid) {
+    __ockl_gws_init(nwm1, rid);
```

Comment:
This kernel needs to initialize resource 0.   Seems to be a waste of cycles on both host and device to set up two arguments and make this handle an arbitrary resource id.

Reply:
Removed the second argument for init_gws kernel.